src/main/java/org/web3j/tx/ManagedTransaction.java

```diff
@@ -75,8 +81,13 @@ protected TransactionReceipt processResponse(EthSendTransaction transactionRespo
         }
 
         String transactionHash = transactionResponse.getTransactionHash();
-
-        return waitForTransactionReceipt(transactionHash);
+        if (attempts > 0) {
```

Comment:
Rather than polling, what about subscribing to the block observable and checking each block to see if it contains the transaction receipt.